components/NavigationBar/examples/NavigationBarTypicalUseExample.swift

```diff
@@ -37,14 +37,34 @@ open class NavigationBarTypicalUseSwiftExample: NavigationBarTypicalUseExample {
 
     navBar!.translatesAutoresizingMaskIntoConstraints = false
 
-    let viewBindings = ["navBar": navBar!]
+    #if swift(>=3.2)
+      if #available(iOS 11.0, *) {
+        self.view.safeAreaLayoutGuide.topAnchor.constraint(equalTo: self.navBar!.topAnchor).isActive = true
+      } else {
+        NSLayoutConstraint(item: self.topLayoutGuide,
```

Comment:
Can you use the same #if #endif as in Objc examples? It should avoid duplicating the non-iOS 11 constraints, yes?

Reply:
I wish; Swift macros don't play as nice:
![screen shot 2017-12-08 at 12 08 39 pm](https://user-images.githubusercontent.com/4066863/33776846-b603be70-dc10-11e7-81c0-11526de31314.png)

Reply:
Must some sort of fake macros. #PreprocessorForever